### PR TITLE
[BROWSEUI] Add CAddressEditBox::RefreshAddress and use it

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Follow-up of #5026 `[BROWSEUI] Execute command line from address bar`.
Elegantly reset the address bar after command line execution.

JIRA issue: [CORE-15453](https://jira.reactos.org/browse/CORE-15453)

## Proposed changes

- Move the code of `CAddressEditBox::Invoke` into newly-defined `CAddressEditBox::RefreshAddress` function.
- Use `CAddressEditBox::RefreshAddress`.

## TODO

- [x] Do tests

### CODE REVIEW

File: dll/win32/browseui/addresseditbox.cpp [hunk 1]
```diff
@@ -133,7 +213,7 @@ BOOL CAddressEditBox::ExecuteCommandLine()
     PathUnquoteSpacesW(pszCmdLine); /* Unquote the 1st parameter */
 
     /* Get ready for execution */
-    SHELLEXECUTEINFOW info = { sizeof(info), SEE_MASK_FLAG_NO_UI, m_hWnd };
+    SHELLEXECUTEINFOW info = { sizeof(info), SEE_MASK_FLAG_NO_UI | SEE_MASK_ASYNCOK, m_hWnd };
```
Comment (learn-more) [inline dll/win32/browseui/addresseditbox.cpp:216]: Why was this changed?

Comment (katahiromz) [reply dll/win32/browseui/addresseditbox.cpp:216]: Is it quicker?

Comment (katahiromz) [reply dll/win32/browseui/addresseditbox.cpp]: Reverted.

**Review by learn-more — APPROVED** [review]

File: dll/win32/browseui/addresseditbox.cpp [hunk 2]
```diff
@@ -104,6 +104,86 @@ BOOL CAddressEditBox::GetComboBoxText(CComHeapPtr<WCHAR>& pszText)
...
+    CComPtr<IShellFolder> psfDesktop;
+    hr = SHGetDesktopFolder(&psfDesktop);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return S_OK;
+
+    hr = psfDesktop->GetDisplayNameOf(absolutePIDL, SHGDN_FORADDRESSBAR, &ret);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return S_OK;
+
+    hr = StrRetToBufW(&ret, absolutePIDL, szName, 4095);
```
Comment (HBelusca) [inline dll/win32/browseui/addresseditbox.cpp:144]: Why not using `_countof(szName)` there? (also why removing one character here, but not at the other StrRetToBufW call below?)

Comment (learn-more) [reply dll/win32/browseui/addresseditbox.cpp:144]: It's better to ask these questions to the original code author?
